lib/bolt/target.rb

```diff
@@ -4,10 +4,10 @@
 
 module Bolt
   class Target
-    attr_reader :uri, :options
+    attr_reader :options
     # CODEREVIEW: this feels wrong. The altertative is threading inventory through the
     # executor to the RemoteTransport
-    attr_accessor :inventory
+    attr_accessor :uri, :inventory
```

Comment:
If uri can be nil, I think we need to make a getter method that will return the `host` and connection information if `target.uri` is called.

Reply:
Why? IMO the `uri` exists to provide a concise method to pass a lot of connection information in a single string from the CLI.

* None of our transports rely on connecting with the URI directly. 
* It seems better to be explicit about when a real uri was passed vs trying to build one there is too much weirdness around uris and escaping.
* Users should be encouraged to use *name* when stringifying a node. That is why we added a name method previously even though it could only be the URI.

Reply:
Ok - those are good points.